### ISSUE TITLE
`aten::fill_` causes H2D sync

### ISSUE THREAD

**Issue by awgu** [pytorch/pytorch#109115]
`aten::fill_` calls into `aten::item`, which causes a H2D sync. Can we implement `fill_` without this sync?

Example repro:
```
a = torch.tensor(0, device="cuda")
x = torch.rand((100,), device="cuda")
amax = torch.max(torch.abs(x))
a.fill_(amax)
```
`amax` is a 0-dim tensor (i.e. with size `torch.Size([])`) as required by `fill_`.

![Screenshot 2023-09-12 at 12 22 31 PM](https://github.com/pytorch/pytorch/assets/31054793/6bcd5b98-9f1b-4201-a1c0-3e8ab957afc6)

Workaround to avoid the H2D sync:
```
a *= 0
a += amax
```

cc @ptrblck

**Comment by janeyx99** [comment]
I changed the _fused_adam and _fused_adamw ops to be able to take in tensors OR scalars (in two different overloads) and in the tensor case, the cuda code converts the tensor to a scalar before calling into the kernel. This would avoid the sync. See https://github.com/pytorch/pytorch/pull/106916 for implementation deets.
